Source/AIPrototype/Private/Controllers/UnitGroupAIController.cpp

```diff
@@ -0,0 +1,152 @@
+#include "Controllers/UnitGroupAIController.h"
+#include "Misc/CommanderBlackboardKeys.h"
+#include "Misc/DelegateHelpers.h"
+#include "AIPrototype/Public/AIPrototype.h"
+#include "BehaviorTree/BlackboardComponent.h"
+#include "Controllers/UnitAIController.h"
+#include "Perception/AIPerceptionComponent.h"
+#include "Perception/AISense_Sight.h"
+#include "Units/UnitBase.h"
+#include "VisualLogger/VisualLogger.h"
+
+AUnitGroupAIController::AUnitGroupAIController()
+{
+	PerceptionComponent = CreateDefaultSubobject<UAIPerceptionComponent>(TEXT("PerceptionComponent"));
+	PerceptionComponent->SetDominantSense(UAISense_Sight::StaticClass());
+}
+
+void AUnitGroupAIController::InitializeControlledUnits(const TArray<AUnitBase*>& Units)
+{
+	for (const auto unit : Units)
+	{
+		m_ControlledUnits.AddUnique(unit);
+	}
+
+	InitBlackboardAlliesNum();
+}
+
+void AUnitGroupAIController::InitBlackboardAlliesNum()
+{
+	GetBlackboardComponent()->SetValueAsInt(CommanderBBKeys::AlliesNum, m_ControlledUnits.Num());
+}
+
+void AUnitGroupAIController::BeginPlay()
+{
+	Super::BeginPlay();
+
+	BindDelegates();
+	RunBehaviorTree(BehaviorTreeAsset);
+}
+
+void AUnitGroupAIController::BindDelegates()
+{
+	PerceptionComponent->OnTargetPerceptionUpdated.AddDynamic(this, &AUnitGroupAIController::OnTargetPerceptionUpdated);
+
+	// consider group movement finished when commander reaches goal location
+	Chain(this, GetPathFollowingComponent()->OnRequestFinished, OnGroupMovementFinishedDelegate, this);
+	OnGroupMovementFinishedDelegate.AddDynamic(this, &AUnitGroupAIController::OnGroupMovementFinished);
```

Comment:
That works because last unit in a group is so named "commander" - leader of the group that controls its movement using this controller as I noted in code and blueprints. Commander controls his movement along with other units in group (see m_ControlledUnits comment), so PathFollowingComponent works as expected. 
AUnitGroupAIController inherited from AUnitAIController because it also controls a unit (commander of the group) so it must have proper team ID, ability to move etc.

Originally I wanted AUnitGroupAIController to be attached to a group and not to possess any pawn, but it appears that PerceptionComponent works only with AIController that possesses a pawn - that's where concept of commander appeared. 
After that proper name for AUnitGroupAIController should be something like AUnitCommanderController but at that point many namings were based on AUnitGroupAIController naming so I left it how it was and placed comments where they were useful.

To test the behavior you can tweak params in EQ_ChooseRoamLocation (GridHalfSize for example) so a group will move in small segments - BT will correctly finish movement tasks and work as expected. 
For better test results you can set unit spawner to spawn only one unit (that'll be commander) or set 200-300cm AcceptanceRadius for movement task - otherwise commander will struggle to reach its destination due to collisions with other units and movement won't be considered finished for a long time (I didn't solve that problem because that was enough for simple group behavior simulation).

Reply:
So commander is a workaround for PerceptionComponent to work? That's probably why we haven't used it. 
What happens when commander get killed? You need to make a new one from other alive unit and spawn new / reposses group controller?
It looks like overcomplication we don't need. And btw, we already have written logic of PerceptionComponent via simple SweepMultiByObjectType